src/rgw/rgw_kafka.cc

```diff
@@ -210,9 +210,16 @@ bool new_producer(connection_t* conn) {
     return false;
   }
 
+  // set message timeout
+  // according to documentation, value of zero will expire the message based on retries.
+  // however, testing with librdkafka v1.6.1 did not expire the message in that case. hence, a value of zero is changed to 1ms
+  constexpr std::uint64_t min_message_timeout = 1;
+  const auto message_timeout = std::max(min_message_timeout, conn->cct->_conf->rgw_kafka_message_timeout);
+  if (rd_kafka_conf_set(conn->temp_conf, "message.timeout.ms", 
```

Comment:
just think more on this, i was wondering do we want to do this only for sync-notification, coz for non-sync notification having larger timeout means if there is a transient kafka downtime which is more than 5 seconds but less than 5 minutes (kafka default), the persistent notification will just wait and succeed eventually vs retrying to send it ?
for sync notification latency is associated, so having 5 seconds there make sense

Reply:
for persistenmt notifications a long timeout creates a different issue, as the coroutines of the notification manager will just pile up and consume significant amount of memory.
i think we did not see that as a huge problem so far, becasue of the 30 seconds idle timeout.
but if set to 5 minutes, it might create an issue.

Reply:
Ohh I see, so then you want to keep the behaviour for persistent same (30 seconds) since we have not seen any issues and seems to working as expected ?
Keeping 5 seconds will now exhaust the retries faster, and we would writing entries back to queue more often then before.

Reply:
> Ohh I see, so then you want to keep the behaviour for persistent same (30 seconds) since we have not seen any issues and seems to working as expected ? Keeping 5 seconds will now exhaust the retries faster, and we would writing entries back to queue more often then before.

will have to test that to see the impact. but i agree that we dont want to see a RADOS spike when the broker is down.
an option would be to keep 2 connections per broker, one for persistent notifications (with 30sec message timeout), and one for sync notifications (with 5sec message timeout)?

Reply:
Why complicate with 2 connections. 
Rather we could default the timeout to 30 seconds for persistent and override that value for sync ?

Reply:
Why complicate with 2 connections. Rather have a default with 30 seconds for persistent and then override that value for sync notification based on conf value

Reply:
persistency is per topic. connection is per broker.
we don't create a connection per topic, so we would keep 2 connections per broker, each with different conf

Reply:
ohh i see, so how we want to handle this ?
test first or just separate out the 2 connections ?

Reply:
i tested it, and could not see any perf hit in cpu or disk.
looked at the `rgw_notify.cc` code, and it actually makes sense that there will be no impact on the kafka timeout:
* when a persistent notification gets an error, it does not do anything to the queue (no RADOS operation). the notification is deleted from the queue only when it gets an ack
* the "max retry" mechanism is memory based only. meaning, if we defined max retries, and get an error we update the count only in memory and not in the queue, so there is no RADOS operation here either

Reply:
thanks for the verification, so the only side effect now is that retries will happen faster compared to earlier which was waiting for the response for 30 seconds before retrying.